internal/operator/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,127 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package serviceaccount
+
+import (
+	"fmt"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/common"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+	"github.com/gardener/etcd-druid/internal/operator/component"
+	"github.com/gardener/etcd-druid/internal/utils"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ErrGetServiceAccount indicates an error in getting the service account resource.
+	ErrGetServiceAccount druidv1alpha1.ErrorCode = "ERR_GET_SERVICE_ACCOUNT"
+	// ErrSyncServiceAccount indicates an error in syncing the service account resource.
+	ErrSyncServiceAccount druidv1alpha1.ErrorCode = "ERR_SYNC_SERVICE_ACCOUNT"
+	// ErrDeleteServiceAccount indicates an error in deleting the service account resource.
+	ErrDeleteServiceAccount druidv1alpha1.ErrorCode = "ERR_DELETE_SERVICE_ACCOUNT"
+)
+
+type _resource struct {
+	client           client.Client
+	disableAutoMount bool
+}
+
+// New returns a new service account operator.
+func New(client client.Client, disableAutomount bool) component.Operator {
+	return &_resource{
+		client:           client,
+		disableAutoMount: disableAutomount,
+	}
+}
+
+// GetExistingResourceNames returns the name of the existing service account for the given Etcd.
+func (r _resource) GetExistingResourceNames(ctx component.OperatorContext, etcd *druidv1alpha1.Etcd) ([]string, error) {
+	resourceNames := make([]string, 0, 1)
+	sa := &corev1.ServiceAccount{}
+	objectKey := getObjectKey(etcd)
+	if err := r.client.Get(ctx, objectKey, sa); err != nil {
```

Comment:
This could be metadata-only get instead of the full object.

Reply:
done